### ISSUE TITLE
Photometry deduplication logic does not exclude record being updated when checking against existing records

### ISSUE THREAD

**Issue by acrellin** [skyportal/skyportal#1908]
A PATCH or PUT request can lead to a duplication error b/c the new data will be checked against the corresponding existing record, leading to an error if none of the values in the deduplication index are being changed.

**Comment by dannygoldstein** [comment]
Yes this is not good, we should allow PK mutations. Currently our order of ops is: 1. Check for duplicates. 2. Mutate.  3. Commit.

Perhaps we can fix this by simply reversing the above? I.e., 1. Mutate 2. Check for duplicates, 3. Commit?

**Comment by mcoughlin** [comment]
@profjsb @guynir42 Do we know understand this well enough to make it actionable?

**Comment by guynir42** [comment]
I think I know what they are talking about.

**Comment by mcoughlin** [comment]
Rewritten to atomic INSERT … ON CONFLICT
